tests/grid_test.py

```diff
@@ -0,0 +1,214 @@
+import numpy
+import pytest
+
+from smolyay.basis import ChebyshevFirstKind
+from smolyay.grid import (IndexGridGenerator, SmolyakGridGenerator,
+                          TensorGridGenerator,
+                          IndexGrid, NestedIndexGrid, generate_compositions)
+
+
+def test_error_basis_set_type():
+    """Test if raise TypeError works for IndexGridGenerator."""
+    basis_set = "not a BasisFunctionSet"
+    with pytest.raises(TypeError):
+        IndexGridGenerator(basis_set)
+
+
+def test_error_nested_basis_set_type():
+    """Test if raise TypeError works for SmolyakGridGenerator."""
+    basis_set = "not a NestedBasisFunctionSet"
+    with pytest.raises(TypeError):
+        SmolyakGridGenerator(basis_set)
+
+
+def test_smolyak_grid_points_indexes_one_dimension():
+    """Test integer grid points for one dimension."""
+    test_nested_basis_set = ChebyshevFirstKind.make_nested_set(2)
+    test_class = SmolyakGridGenerator(test_nested_basis_set)
+    expected_integer_grid_points = [[0], [1], [2], [3], [4]]
+    assert (test_class(1).indexes
+            == expected_integer_grid_points)
+
+
+def test_smolyak_grid_points_indexes():
+    """Test integer grid points."""
+    test_nested_basis_set = ChebyshevFirstKind.make_nested_set(2)
+    test_class = SmolyakGridGenerator(test_nested_basis_set)
+    expected_integer_grid_points = [[0, 0], [1, 0], [2, 0],
+                                    [0, 1], [0, 2], [3, 0], [4, 0],
+                                    [1, 1], [1, 2], [2, 1],
+                                    [2, 2], [0, 3], [0, 4]]
+    assert (test_class(2).indexes
+            == expected_integer_grid_points)
+
+
+def test_smolyak_levels_one_dimension():
+    """Test levels of the grid points. in one dimension."""
+    test_nested_basis_set = ChebyshevFirstKind.make_nested_set(3)
+    test_class = SmolyakGridGenerator(test_nested_basis_set)
+    expected_levels = [[0], [1], [1], [2], [2], [3], [3], [3], [3]]
+    assert test_class(1).levels == expected_levels
+
+
+def test_smolyak_levels():
```

Comment:
It might make sense to test the indexes, levels, etc. all in one test function, so you don't have to jump around as much reading. It's OK for tests to make multiple assertions.